pkgs/by-name/op/ophcrack-cli/package.nix

```
@@ -0,0 +1,54 @@
+{ lib
+, stdenvNoCC
+, libsForQt5
+, fetchFromGitLab
+, autoreconfHook
+, libtool
+, pkg-config
+, zlib
+, openssl
+, freetype
+, fontconfig
+, gcc
+, expat
+, cmake
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "ophcrack";
+  version = "3.8.0-3";
+
+  src = fetchFromGitLab {
+    domain = "salsa.debian.org";
+    owner = "pkg-security-team";
+    repo = "ophcrack";
+    rev = "debian/${version}";
+    hash = "sha256-94zEr7aBeqMjy5Ma50W3qv1S5yx090bYuTieoZaXFcc=";
+  };
+  nativeBuildInputs = [ autoreconfHook libtool libsForQt5.wrapQtAppsHook ];
+  buildInputs = [ pkg-config zlib openssl freetype fontconfig libsForQt5.qtcharts gcc expat ];
+  dontWrapQtApps = true;
+  configureFlags = [ "CFLAGS=-I/usr/include/libxml2/libxml, --disable-gui" ];
+  buildPhase = ''
+    make
+  '';
```

```suggestion

  nativeBuildInputs = [ 
    autoreconfHook 
    libtool 
    libsForQt5.wrapQtAppsHook 
  ];

  buildInputs = [ 
    pkg-config 
    zlib 
    openssl 
    freetype 
    fontconfig 
    libsForQt5.qtcharts 
    gcc 
    expat 
  ];

  dontWrapQtApps = true;
  
  configureFlags = [ 
    "--disable-gui" 
  ];

  env.CFLAGS = toString [ 
    "-I/usr/include/libxml2/libxml" 
  ];

```

P.S.: why GCC and  pkg-config?

---

packaging the software without GCC will result in error, and the same applies when using pkg-config alone but when both GCC and pkg-config are used the software compiles correctly.